From dcbd56c70a9929011095ab1e24fa5f8d40b0a258 Mon Sep 17 00:00:00 2001
From: Huang Shan <huang.shan@freescale.com>
Date: Thu, 8 May 2014 12:53:13 +0800
Subject: [PATCH 64/70] ENGR00303833-1 ethernet:clear proxy when unset it.

Shutdowm the proxy when the proxy hostname or port is null

Signed-off-by: Huang Shan <huang.shan@freescale.com>
---
 Ethernet/com/fsl/ethernet/EthernetAdvDialog.java | 2 +-
 Ethernet/com/fsl/ethernet/EthernetManager.java   | 4 +++-
 2 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/Ethernet/com/fsl/ethernet/EthernetAdvDialog.java b/Ethernet/com/fsl/ethernet/EthernetAdvDialog.java
index e6473da..d0b7728 100644
--- a/Ethernet/com/fsl/ethernet/EthernetAdvDialog.java
+++ b/Ethernet/com/fsl/ethernet/EthernetAdvDialog.java
@@ -90,7 +90,7 @@ public class EthernetAdvDialog extends AlertDialog implements DialogInterface.On
             info.setProxyExclusionList(mProxyExclusionList.getText().toString());
         }
         mEthEnabler.getManager().updateDevInfo(info);
-        mEthEnabler.getManager().setProxy( );
+        mEthEnabler.getManager().setProxy();
     }
 
     public void onClick(DialogInterface dialog, int which) {
diff --git a/Ethernet/com/fsl/ethernet/EthernetManager.java b/Ethernet/com/fsl/ethernet/EthernetManager.java
index 8ba9a43..bde153c 100644
--- a/Ethernet/com/fsl/ethernet/EthernetManager.java
+++ b/Ethernet/com/fsl/ethernet/EthernetManager.java
@@ -365,8 +365,10 @@ public class EthernetManager {
     public void setProxy(){
         String exclusionList = null;
         exclusionList=getSharedPreProxyExclusionList();
-        if (getSharedPreProxyAddress() == null || getSharedPreProxyPort() == null)
+        if (getSharedPreProxyAddress() == null || getSharedPreProxyPort() == null) {
+            mConnMgr.setGlobalProxy(null);
             return;
+        }
         LinkProperties lp = mTracker.getLinkProperties();
         if (lp == null)
             return;
-- 
1.8.0

